### PR TITLE
Docs: Update canvas docs for 9.4

### PR DESCRIPTION
**Pull request by nmarrs**
Update docs for 9.4 for canvas

TODO:

- [x] Add video demos
- [x] Review all of canvas changes to ensure that docs are up to date as possible

### CODE REVIEW

File: docs/sources/panels-visualizations/visualizations/canvas/index.md [hunk 1]
```diff
@@ -42,6 +42,12 @@ The rectangle element enables you to add a basic rectangle to the canvas. Rectan
 
 The icon element enables you to add a supported icon to the canvas. Icons can have their color set based on thresholds / value mappings.
 
+## Connections
+
+When building a canvas panel, you can connect elements together to create more complex visualizations. Connections are created by dragging from the connection anchor of one element to the connection anchor of another element. Connections can also be created to the background of the canvas panel. Connection anchors are displayed when an element is hovered over and inline editing is turned on. To remove a connection, simply click on the connection directly and then press the "Delete" key.
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/canvas/index.md:47]: Removing some of the passive voice

```suggestion
When building a canvas panel, you can connect elements together to create more complex visualizations. Connections are created by dragging from the connection anchor of one element to the connection anchor of another element. You can also create connections to the background of the canvas panel. Connection anchors are displayed when you hover over an element and inline editing is turned on. To remove a connection, simply click on the connection directly and then press the "Delete" key.
```

File: docs/sources/panels-visualizations/visualizations/canvas/index.md [hunk 2]
```diff
@@ -42,6 +42,12 @@ The rectangle element enables you to add a basic rectangle to the canvas. Rectan
...
+
+{{< video-embed src="/static/img/docs/canvas-panel/canvas-connections-9-4-0.mp4" max-width="750px" caption="Canvas connections demo" >}}
+
 ## Canvas Editing
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/canvas/index.md:51]:
```suggestion
## Canvas editing
```

File: docs/sources/panels-visualizations/visualizations/canvas/index.md [hunk 3]
```diff
@@ -69,3 +75,14 @@ When right clicking an element, you are able to edit, delete, duplicate, and mod
 The inline editing toggle enables you to lock or unlock the canvas panel. When turned off the canvas panel becomes “locked”, freezing elements in place and preventing unintended modifications.
 
 {{< video-embed src="/static/img/docs/canvas-panel/canvas-inline-editing-toggle-9-2-0.mp4" max-width="750px" caption="Inline editing toggle demo" >}}
+
+### Data Links
+
+Canvas supports [data links](https://grafana.com/docs/grafana/latest/panels-visualizations/configure-data-links/). Once you have added a data link to the panel, you can display it following these steps:
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/canvas/index.md:81]:
```suggestion
Canvas supports [data links](https://grafana.com/docs/grafana/latest/panels-visualizations/configure-data-links/). Once you've added a data link to the panel, you can display it by following these steps:
```

File: docs/sources/panels-visualizations/visualizations/canvas/index.md [hunk 4]
```diff
@@ -69,3 +75,14 @@ When right clicking an element, you are able to edit, delete, duplicate, and mod
...
+
+1. Set an element to be tied to a field value
+2. Turn off the inline editing toggle
+3. Now hover over the element and a data link tooltip will appear
+4. Click the data link tooltip to open the data link
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/canvas/index.md:86]: Adding missing punctuation and a few wording edits

```suggestion
1. Set an element to be tied to a field value.
2. Turn off the inline editing toggle.
3. Hover over the element to display the data link tooltip.
4. Click the data link tooltip to open the data link.
```

File: docs/sources/panels-visualizations/visualizations/canvas/index.md [hunk 5]
```diff
@@ -69,3 +75,14 @@ When right clicking an element, you are able to edit, delete, duplicate, and mod
 The inline editing toggle enables you to lock or unlock the canvas panel. When turned off the canvas panel becomes “locked”, freezing elements in place and preventing unintended modifications.
 
 {{< video-embed src="/static/img/docs/canvas-panel/canvas-inline-editing-toggle-9-2-0.mp4" max-width="750px" caption="Inline editing toggle demo" >}}
+
+### Data Links
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/canvas/index.md:79]:
```suggestion
### Data links
```

**Review by imatwawana — COMMENTED** [review]
I've made some styling and wording edits

File: docs/sources/panels-visualizations/visualizations/canvas/index.md [hunk 6]
```diff
@@ -69,3 +75,14 @@ When right clicking an element, you are able to edit, delete, duplicate, and mod
...
 {{< video-embed src="/static/img/docs/canvas-panel/canvas-inline-editing-toggle-9-2-0.mp4" max-width="750px" caption="Inline editing toggle demo" >}}
+
+### Data links
+
+Canvas supports [data links](https://grafana.com/docs/grafana/latest/panels-visualizations/configure-data-links/). Once you've added a data link to the panel, you can display it by following these steps:
+
+1. Set an element to be tied to a field value.
+2. Turn off the inline editing toggle.
+3. Hover over the element to display the data link tooltip.
+4. Click the data link tooltip to open the data link.
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/canvas/index.md:86]:
```suggestion
1. Set an element to be tied to a field value.
1. Turn off the inline editing toggle.
1. Hover over the element to display the data link tooltip.
1. Click the data link tooltip to open the data link.
```

**Review by imatwawana — APPROVED** [review]
I forgot to simplify the numeration, but otherwise, this LGTM!

File: docs/sources/panels-visualizations/visualizations/canvas/index.md [hunk 7]
```diff
@@ -42,7 +42,19 @@ The rectangle element enables you to add a basic rectangle to the canvas. Rectan
...
-## Canvas Editing
+### Server
+
+The server element enables you to easily represent a single server, a stack of servers, a database, or a terminal. Server elements support status color, bulb color, and a bulb blink rate all configurable by fixed or field values.
+
+{{< figure src="/media/docs/grafana/canvas-server-element-9-4-0.png" max-width="750px" caption="Canvas server element" >}}
+
+## Connections
+
+When building a canvas panel, you can connect elements together to create more complex visualizations. Connections are created by dragging from the connection anchor of one element to the connection anchor of another element. You can also create connections to the background of the canvas panel. Connection anchors are displayed when you hover over an element and inline editing is turned on. To remove a connection, simply click on the connection directly and then press the "Delete" key.
```
Comment (drew08t) [inline docs/sources/panels-visualizations/visualizations/canvas/index.md:53]: Delete key or backspace, right?

Comment (nmarrs) [reply docs/sources/panels-visualizations/visualizations/canvas/index.md:53]: yes - do you think it is worth mentioning both?

File: docs/sources/panels-visualizations/visualizations/canvas/index.md [hunk 8]
```diff
@@ -62,10 +74,21 @@ When right clicking an element, you are able to edit, delete, duplicate, and mod
...
 The inline editing toggle enables you to lock or unlock the canvas panel. When turned off the canvas panel becomes “locked”, freezing elements in place and preventing unintended modifications.
 
 {{< video-embed src="/static/img/docs/canvas-panel/canvas-inline-editing-toggle-9-2-0.mp4" max-width="750px" caption="Inline editing toggle demo" >}}
+
+### Data links
+
+Canvas supports [data links](https://grafana.com/docs/grafana/latest/panels-visualizations/configure-data-links/). Once you've added a data link to the panel, you can display it by following these steps:
+
+1. Set an element to be tied to a field value.
+1. Turn off the inline editing toggle.
```
Comment (drew08t) [inline docs/sources/panels-visualizations/visualizations/canvas/index.md:90]: Are these numbers supposed to increment?

Comment (nmarrs) [reply docs/sources/panels-visualizations/visualizations/canvas/index.md:90]: I believe they auto increment - [see this comment](https://github.com/grafana/grafana/pull/62923#discussion_r1100398712)

File: docs/sources/panels-visualizations/visualizations/canvas/index.md [hunk 9]
```diff
@@ -62,10 +74,21 @@ When right clicking an element, you are able to edit, delete, duplicate, and mod
...
+1. Hover over the element to display the data link tooltip.
+1. Click on the element to be able to open the data link.
+
+{{< video-embed src="/media/docs/grafana/canvas-data-links-9-4-0.mp4" max-width="750px" caption="Data links demo" >}}
```
Comment (drew08t) [inline docs/sources/panels-visualizations/visualizations/canvas/index.md:94]: Are embedded videos supported in docs now? If so, cool!

File: docs/sources/panels-visualizations/visualizations/canvas/index.md (hunk 7 above)
Comment (drew08t) [reply docs/sources/panels-visualizations/visualizations/canvas/index.md:53]: Do we have a tooltip when a connection is highlighted that explicitly provides these instructions? If not, I think it is worth mentioning both in the docs for completeness.

File: docs/sources/panels-visualizations/visualizations/canvas/index.md (hunk 8 above)
Comment (drew08t) [reply docs/sources/panels-visualizations/visualizations/canvas/index.md:90]: Nice!

**Pull request merged**